src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -1440,6 +1489,18 @@ public override DeclarationScope Scope(int index)
             return _parameterScopes.IsDefault ? DeclarationScope.Unscoped : _parameterScopes[index];
         }
 
+        public override EqualsValueClauseSyntax? DefaultValue(int index)
+        {
+            Debug.Assert(_defaultValues.IsDefault || (0 <= index && index < _defaultValues.Length));
```

Comment:
>_defaultValues.IsDefault || (0 <= index && index < _defaultValues.Length) [](http://example.com/codeflow?start=25&length=73)

Consider strengthening the assert, and matching the other methods above, by comparing `index` with `_parameterTypesWithAnnotations.Length` unconditionally.